src/helm/benchmark/scenarios/mmlu_pro.py

```diff
@@ -0,0 +1,93 @@
+import csv
+import os
+from typing import Dict, List
+
+from helm.common.general import ensure_file_downloaded, ensure_directory_exists
+from helm.common.hierarchical_logger import hlog
+from .scenario import Scenario, Instance, Reference, TRAIN_SPLIT, VALID_SPLIT, TEST_SPLIT, CORRECT_TAG, Input, Output
+
+import datasets
+from typing import List, Dict
+from pathlib import Path
+import json
+import random 
+
+
+from helm.benchmark.scenarios.scenario import (
+    Scenario,
+    Instance,
+    Reference,
+    TEST_SPLIT,
+    Input,
+    Output,
+)
+from helm.common.general import ensure_directory_exists
+
+class MMLUPROScenario(Scenario):
+    """
+    This is sample code for the MMLU-PRO Dataset.
+    """
+
+    name = "mmlu_pro"
+    description = "Massive Multitask Language Understanding PRO"
+    tags = ["knowledge", "multiple_choice"]
+
+    def __init__(self, subject: str):
+        super().__init__()
+        self.subject: str = subject
+
+    def process_csv(self, csv_path: str, split: str) -> List[Instance]:
+        instances: List[Instance] = []
+        hlog(f"Reading {csv_path}")
+        with open(csv_path) as f:
+            reader = csv.reader(f, delimiter=",")
+            for row in reader:
+                # Example: ["What color is the sky?", "red", "blue", "green", "B"]
+                question, answers, correct_choice = row[0], row[1:-1], row[-1]
+                answers_dict = dict(zip(["A", "B", "C", "D", "E"], answers))
+                correct_answer: str = answers_dict[correct_choice]
+
+                def answer_to_reference(answer: str) -> Reference:
+                    return Reference(Output(text=answer), tags=[CORRECT_TAG] if answer == correct_answer else [])
+
+                instance = Instance(
+                    input=Input(text=question),
+                    references=list(map(answer_to_reference, answers)),
+                    split=split,
+                )
+                instances.append(instance)
+        return instances
+
+    def get_instances(self, output_path: str) -> List[Instance]:
+        fields, _, label_key, _ = self.load_prompt_construction_settings(output_path)
+        cache_dir = str(Path(output_path) / "data")
+
+        train_dataset = datasets.load_dataset(
+            "TIGER-Lab/MMLU-Pro", self.subset, trust_remote_code=True, cache_dir=cache_dir, split="train"
```

Comment:
nit: pin the revision like so:

https://github.com/stanford-crfm/helm/blob/e4f2b1ed809f308aa6e69b1768b9fa47e95733bf/src/helm/benchmark/scenarios/ewok_scenario.py#L74

This ensure that we use a fixed version of the dataset even if the authors update the dataset later.